File: 06-june/01-minimum-cost-of-buying-candies-with-discount/solution.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minimumCost = function(cost) {
    cost.sort((a, b) => b - a);

    let answer = 0;
    for (let i = 0; i < cost.length; i++) {
        if (i % 3 !== 2) {
            answer += cost[i];
        }
    }

    return answer;
};

File: 06-june/01-minimum-cost-of-buying-candies-with-discount/solution.java
import java.util.Arrays;

class Solution {
    public int minimumCost(int[] cost) {
        Arrays.sort(cost);

        int answer = 0;
        for (int i = cost.length - 1, taken = 0; i >= 0; i--, taken++) {
            if (taken % 3 != 2) {
                answer += cost[i];
            }
        }

        return answer;
    }
}
